Plugin~/WebRTCPlugin/GraphicsDevice/Vulkan/VulkanGraphicsDevice.cpp

```diff
@@ -143,28 +160,56 @@ bool VulkanGraphicsDevice::CopyResourceV(ITexture2D* dest, ITexture2D* src) {
 }
 
 //---------------------------------------------------------------------------------------------------------------------
-bool VulkanGraphicsDevice::CopyResourceFromNativeV(ITexture2D* dest, void* nativeTexturePtr) {
+bool VulkanGraphicsDevice::CopyResourceFromNativeV(
+    ITexture2D* dest, void* nativeTexturePtr) {
     if (nullptr == dest || nullptr == nativeTexturePtr)
         return false;
 
     VulkanTexture2D* destTexture = reinterpret_cast<VulkanTexture2D*>(dest);
-    UnityVulkanImage unityVulkanImage;
     VkImageSubresource subResource { VK_IMAGE_ASPECT_COLOR_BIT, 0, 0 };
+    VkImage image;
 
-    if (!m_unityVulkan->AccessTexture(nativeTexturePtr, &subResource, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
-        VK_PIPELINE_STAGE_TRANSFER_BIT, VK_ACCESS_TRANSFER_READ_BIT, kUnityVulkanResourceAccess_PipelineBarrier,
-        &unityVulkanImage))
+    // The nativeTexturePtr is texture pointer provided by Unity.
+    // When running native test, this pointer as UnityVulkanImage pointer.
+    if (m_unityVulkan != nullptr)
     {
-        return false;
+        UnityVulkanImage unityVulkanImage;
+        if (!m_unityVulkan->AccessTexture(
+            nativeTexturePtr, &subResource, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+            VK_PIPELINE_STAGE_TRANSFER_BIT, VK_ACCESS_TRANSFER_READ_BIT,
+            kUnityVulkanResourceAccess_PipelineBarrier,
+            &unityVulkanImage))
+        {
+            return false;
+        }
+        image = unityVulkanImage.image;
     }
+    else
+    {
+        UnityVulkanImage* unityVulkanImage = static_cast<UnityVulkanImage*>(nativeTexturePtr);
```

Comment:
The method clearly says CopyResourceFromNativeV, so nativeTexturePtr should be a native texture pointer, not a pointer to UnityVulkanImage.
I don't think we should make the code more complicated, just because we want to test it.
And because there is a branching here, we are not testing the actual runtime code anyway

Is is possible to pass the native texture directly from the testing code instead ?

Reply:
Yeah, you are right. I am also sure that we should not make this branching.
But the type of a native texture pointer is not public from Unity, so we cannot create the texture without Unity.

As an alternative idea, that is defining the texture pointer of argument of this method as `UnityVulkanImage`.
How about this idea?

Reply:
But  you want to keep the name as CopyResourceFromNativeV(), or do you want to create a new method ?

Reply:
Hm, not ideal and might be misleading, but I see that this might be a good trade off rather than changing the structure of the code.
Ok.

Reply:
fixed